main.ts

```diff
@@ -84,7 +87,77 @@ export default class LetterboxdPlugin extends Plugin {
 					})
 			},
 		})
-
+		this.addCommand({
+			id:"add-last-movie",
+			name: "Add Last watched movie",
+			callback: async () => {
+			if (!this.settings.username) {
+				throw new Error('Cannot get data for blank username')
+			}
+			requestUrl(`https://letterboxd.com/${this.settings.username}/rss/`)
+				.then(res => res.text)
+				.then(async res => {
+					const parser1 = new XMLParser();
+					let jObj1 = parser1.parse(res);
+					let item = jObj1.rss.channel.item[0] 
+					this.app.workspace.activeEditor?.editor?.replaceRange(
+						`>[!Last Movie Logged]+ \n> [${item['title']}](${item['link']}) on ${item['letterboxd:watchedDate']} \n> ${item['description']}  ` ,
+						this.app.workspace.activeEditor.editor.getCursor()
+					);
+				})
+			},
+		})
+		this.addCommand({
+			id:"add-movies-watched-today",
+			name:"Add Movies watched today",
+			callback: async () => {
+			if (!this.settings.username) {
+				throw new Error('Cannot get data for blank username')
+			}
+			requestUrl(`https://letterboxd.com/${this.settings.username}/rss/`)
+				.then(res => res.text)
+				.then(async res => {
+					const parser1 = new XMLParser();
+					let jObj1 = parser1.parse(res);
+					let items = jObj1.rss.channel.item 
+					let d = new Date()
+					d.setHours(0,0,0,0)
+					let movies_today=items.filter(entry => Date.parse(entry.pubDate) > d);
+					var text = ">[!Movies added on letterboxd today]+ \n"
+					for (var item of movies_today) {
+						text=text+`> [${item['title']}](${item['link']}) on ${item['letterboxd:watchedDate']} \n> ${item['description']}  `
+					}					
+					this.app.workspace.activeEditor?.editor?.replaceRange(
+						text,
+						this.app.workspace.activeEditor.editor.getCursor()
+					);
+				})
+			},
+		})
+		this.addCommand({
+			id:"add-last-2-movies",
+			name: "Add Last 2 watched movie",
+				callback: async () => {
+				if (!this.settings.username) {
+					throw new Error('Cannot get data for blank username')
+				}
+				requestUrl(`https://letterboxd.com/${this.settings.username}/rss/`)
+					.then(res => res.text)
+					.then(async res => {
+						const parser1 = new XMLParser();
+						let jObj1 = parser1.parse(res);
+						let items = [jObj1.rss.channel.item[0],jObj1.rss.channel.item[1]] 
+						var text = ">[!Last two Movies Logged]+ \n"
```

Comment:
You're using a bit of the same logic for all three commands and it could probably be refactored